sdk/communication/communication-job-router/src/routerClient.ts

```diff
@@ -0,0 +1,673 @@
+  private readonly client: JobRouterApiClient;
+
+  /**
+   * Initializes a new instance of the RouterClient class.
+   * @param connectionString - Connection string to connect to an Azure Communication Service resource.
+   *                         Example: "endpoint=https://contoso.eastus.communications.azure.net/;accesskey=secret";
+   * @param options - Optional. Options to configure the HTTP pipeline.
+   */
+  constructor(connectionString: string, options?: RouterClientOptions);
+
+  /**
+   * Initializes a new instance of the RouterClient class using an Azure KeyCredential.
+   * @param endpoint - The endpoint of the service (ex: https://contoso.eastus.communications.azure.net).
+   * @param credential - An object that is used to authenticate requests to the service. Use the Azure KeyCredential or `@azure/identity` to create a credential.
+   * @param options - Optional. Options to configure the HTTP pipeline.
+   */
+  constructor(endpoint: string, credential: KeyCredential, options?: RouterClientOptions);
+
+  /**
+   * Initializes a new instance of the RouterClient class using a TokenCredential.
+   * @param endpoint - The endpoint of the service (ex: https://contoso.eastus.communications.azure.net).
+   * @param credential - TokenCredential that is used to authenticate requests to the service.
+   * @param options - Optional. Options to configure the HTTP pipeline.
+   */
+  constructor(endpoint: string, credential: TokenCredential, options?: RouterClientOptions);
+
+  /**
+   * Creates an instance of the RouterClient for a given resource and user.
+   *
+   * @param connectionStringOrUrl - The connectionString or url of the Communication Services resource.
+   * @param credentialOrOptions - The key or token credential or RouterClientOptions. Use AzureCommunicationKeyCredential from \@azure/communication-common to create a credential.
+   * @param maybeOptions - Additional client options.
+   */
+  constructor(
+    connectionStringOrUrl: string,
+    credentialOrOptions?: KeyCredential | TokenCredential | RouterClientOptions,
+    maybeOptions: RouterClientOptions = {}
+  ) {
+    const { url, credential } = parseClientArguments(connectionStringOrUrl, credentialOrOptions);
+    const options = isRouterClientOptions(credentialOrOptions) ? credentialOrOptions : maybeOptions;
+
+    const libInfo = `azsdk-js-communication-job-router/${SDK_VERSION}`;
+
+    if (!options.userAgentOptions) {
+      options.userAgentOptions = {};
+    }
+
+    const userAgentOptions = { ...options.userAgentOptions };
+    if (options.userAgentOptions.userAgentPrefix) {
+      userAgentOptions.userAgentPrefix = `${options.userAgentOptions.userAgentPrefix} ${libInfo}`;
+    } else {
+      userAgentOptions.userAgentPrefix = libInfo;
+    }
+
+    const internalPipelineOptions: InternalPipelineOptions = {
+      ...{ ...options, userAgentOptions },
+      ...{
+        loggingOptions: {
+          logger: logger.info
+        }
+      }
+    };
```

Comment:
```suggestion
    const internalPipelineOptions: InternalPipelineOptions = {
      ...options,
      userAgentOptions,
      loggingOptions: {
        logger: logger.info
      }
    };
```

What about this? Seems to be about the same.